### ISSUE TITLE
Add device selection via getSources to Apprtc

### ISSUE THREAD

**Issue by jiayliu** [webrtc/apprtc#8]
https://github.com/GoogleChrome/webrtc/issues/398

Not sure if this has been requested before but I think it makes sense to add device selection via JS saving the user from reloading the tab in order to change device.

samdutton commented 15 days ago
There was some discussion of this apropos of #311: the decision was to handle this in a separate PR.

Maybe should wait until RtpSender.replaceTrack() is implemented, though could do a short term hack (to handle the change of device at the 'other end' of the call)?

@juberti
Justin Uberti
 Owner
juberti commented 14 days ago
The short-term way to do this is to do SDP munging to change the MSID on the local description. But it might just be easiest to wait for replaceTrack; we're trying to get to that this quarter.
Sam Dutton
 Owner
 samdutton commented 14 days ago
I agree: let's wait.
…
Christoffer Jansson
 Owner
KaptenJansson commented 9 days ago
Also, it would be nice if this could be added to the new landing page as well, maybe as a first step as then you would not need to handle renegotiation etc.

**Comment by KaptenJansson** [comment]
Do we still want to add device selection (soon output as well?) to apprtc? @samdutton @juberti WDYT?

**Comment by KaptenJansson** [comment]
Ping?

**Comment by samdutton** [comment]
I'd say yes, if we can keep the UI/UX simple.

IIRC the original version had a camera chooser, but I think we removed this
until the device swap could be automatically handled, rather than
initiating a new call.

**Comment by juberti** [comment]
I think we should do this, but only once we have RtpSender.replaceTrack. People desperately want to see the recommended way to switch cameras.

**Comment by FluorescentHallucinogen** [comment]
Any progress? Choose camera and microphone input sources and speaker output destination using UI is "must have" features for this app. All these things are already implemented in https://webrtc.github.io/samples/src/content/devices/input-output/ demo.

**Comment by KaptenJansson** [comment]
Not sure about must have but the reason why it's not implemented yet is in the comment [above](https://github.com/webrtc/apprtc/issues/8#issuecomment-187435808).

**Comment by kulikov-e-s** [comment]
What about changing cameras on the phone during the call?

**Comment by KaptenJansson** [comment]
Closing due to age.
